src/Uplink/Admin/Provider.php

```diff
@@ -62,6 +63,18 @@ public function filter_plugins_api( $result, $action, $args ) {
 		return $this->container->get( Plugins_Page::class )->inject_info( $result, $action, $args );
 	}
 
+	/**
+	 * Handle auth and disconnect requests to origin
+	 *
+	 * @param \WP $wp
+	 */
+	public function auth_request( $wp ) {
+		if ( ! is_user_logged_in() ) {
```

Comment:
instead of only logged in users, it might be a good idea to also check user capabilities `current_user_can( 'manage_options' )`